snowplow-tracker/src/main/java/com/snowplowanalytics/snowplow/internal/session/Session.java

```diff
@@ -258,35 +258,28 @@ public void startNewSession() {
     }
 
     /**
-     * Updates the session with information about lifecycle tracking.
+     * Updates the session timeout and the indexes.
      * Note: Internal use only.
-     * @param isForeground whether or not the application moved to foreground.
-     * @return foreground or background index. Returns -1 if the lifecycle state is not changed.
+     * @param isBackground whether or not the application moved to background.
      */
-    public synchronized int updateLifecycleNotification(boolean isForeground) {
-        boolean toBackground = !isForeground;
-        // if the new lifecycle state confirms the session state, there isn't any lifecycle transition
-        if (isBackground.get() == toBackground) {
-            return -1;
+    public void setBackground(boolean isBackground) {
```

Comment:
I like that you renamed `updateLifecycleNotification` function to `setBackground`, was quite confusing to me before! But not sure why isn't it synchronized anymore? Is it no longer necessary?

Reply:
I was in doubt but now this method is simplified. The unique operations of writing are on booleans and integer. `isBackground` is an AtomicBoolean, the others are just set and get (no compare) so shouldn't need any synchronization. However, I think it would be good to make them `volatile` in order to pass the info among threads.